### DISCUSSION TITLE
Why is the spectral distribution recovery optimization failing?

### DISCUSSION THREAD

**Discussion by YunEr-Wang** [colour-science/colour#873] [Q&A] [answered]
Dear author,
Recently, I used the color package to recovery the spectral distribution from XYZ values. However, the optimization is usually failed when I set the method to "Meng 2015". Even if I increase the max iteration number, the result is the same. Could you please help me with it? Thank you. 
```
XYZ="0.000000e+00 2.933989e-01 2.933989e-01"
RuntimeError: Optimization failed for [ 0.         0.2933989  0.2933989] after 100 iterations: "Iteration limit reached".
```

**Comment by thomasmansencal** [comment, marked as answer]
Hi @Naughty-Monkey,

Thanks, this is a good question! The reason is simply, the tristimulus values you are using are outside the spectral locus / visible spectrum, thus they cannot be recovered:

![image](https://user-images.githubusercontent.com/99779/133420959-d382abb7-1b08-4777-8fe4-7ad259d96f95.png)

This is especially true for Meng (2015) as it tries to generate the smoothest reflectance for given tristimulus values while your coordinates are physically non-realisable and would require a bandwidth beyond the thinnest laser.

**Reply by YunEr-Wang** [reply]
Yeah, you are correct. It can work when the point is visible in the color space.

**Reply by thomasmansencal** [reply]
If you want to reproduce the figure:

```python
>>> import colour
>>> import matplotlib.pyplot as plt
>>> XYZ = [0., 0.2933989, 0.2933989]
>>> xy = colour.XYZ_to_xy(XYZ)
>>> colour.plotting.plot_chromaticity_diagram_CIE1931(standalone=False)
(<Figure size 640x640 with 1 Axes>, <AxesSubplot:title={'center':'CIE 1931 Chromaticity Diagram - CIE 1931 2$^\\circ$ Standard Observer'}, xlabel='CIE x', ylabel='CIE y'>)
>>> plt.plot(xy[0], xy[1], 'o')
[<matplotlib.lines.Line2D object at 0x12bcd86a0>]
>>> colour.plotting.render(bounding_box=[-.1, .9, -.1, .9])
(<Figure size 1280x1280 with 1 Axes>, <AxesSubplot:title={'center':'CIE 1931 Chromaticity Diagram - CIE 1931 2$^\\circ$ Standard Observer'}, xlabel='CIE x', ylabel='CIE y'>)
```

There is also the `colour.is_within_visible_spectrum` definition that can help you here! To go back to Meng (2015), it is also worth noting that you won't be able to recover spectrum close or on the spectral locus. They are typically extremely narrow-band.
